docs/design/job_architecture.md

```diff
@@ -25,167 +65,15 @@ define(['common/jobCommMessages'], (jcm) => {
-
-### Job-related
-
-`INFO` - contains information about the current job. The format is described under `job_info` in the **Messages sent from the kernel to the browser** section below.
-
-`LOGS` - sent with information about some job logs. The format is described under `job_logs` in the **Messages sent from the kernel to the browser** section below.
-
-`RETRY` - sent in response to a retry request; contains the original job ID, the ID of the retry if the request was successful, or an error message, if it was not.
-
-`STATUS` - contains the current job state
-  * `jobState` - object, describes the job state (see the **Data Structures** section below for the structure)
-  * `outputWidgetInfo` - object, contains the parameters to be sent to an output widget. This will be different for all widgets, depending on the App that invokes them.
-
-`ERROR` - sent in response to an error that happened on job information lookup, or another error that happened while processing some other message to the JobManager.
-  * `raw_request` - object; the request that generated the error
-  * `source` - string; what triggered the error, e.g. an invalid request or an issue completing a request
-  * any other content that was supplied - this is frequently an error object, from which a name and error message, and potentially other information, are taken.
-
-
-### Usage example
-As in the Bus requests section above, the front end response handling is done through the Runtime bus. The bus provides both an `on` and a `listen` function, examples will show how to use both. Generally, the `listen` function is more specific and binds the listener to a specific bus channel. These channels can invoke the jobId, or the cellId, to make sure that only information about specific jobs is listened for.
-
-The `listen` function takes an object with three attributes as input - a channel (either the cellId or jobId), a key (with the type of message to listen for), and a handle, which is a function to process the message. This is probably the easiest way to handle messages. Usage would look like this:
-
-```Javascript
-define(
-  ['common/runtime', 'common/jobCommMessages'],
-  function(Runtime, jcm) {
-    const runtime = Runtime.make();
-
-    const listener = runtime.bus().listen({
-      channel: {
-        [jcm.CHANNEL.JOB]: 'some_job_id'
-      },
-      key: {
-        type: jcm.MESSAGE_TYPE.STATUS,
-      },
-      handle: (message) => {
-        ...process the message...
-      }
-    })
-  }
-);
-```
-
-The `on` function requires a constructed channel bus, premade and reusable for a given channel. So you would make a channel bus that would always receive messages for that channel, and instruct it on what to do when a message of a given type arrives. That looks like this:
-```Javascript
-define(
-  ['common/runtime', 'common/jobCommMessages'],
-  function(Runtime, jcm) {
-    const runtime = Runtime.make();
+## Messages from frontend to backend

```

Comment:
Now that the front and backend use the same vocabulary for job messages, there is a lot of redundancy in this documentation. I've combined the repeated material and reorganised the docs so they are (I hope) a bit simpler and far shorter!